pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,343 @@
+
+			if err != nil {
+				return fmt.Errorf("error getting roles: %v", err)
+			}
+
+			var role *rbac.ClusterRole
+
+			for _, candidate := range roles {
+				if candidate.Annotations[podSecurityAnnotation] == "true" {
+					role = candidate
+				}
+			}
+
+			if role == nil {
+				// Create role
+				newRole := &rbac.ClusterRole{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{},
+						Name:        fmt.Sprintf("%v-clusterrole", policy.Name),
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind: "ClusterRole",
+					},
+					Rules: []rbac.PolicyRule{
+						{
+							APIGroups:     []string{"extensions"},
+							Resources:     []string{"podsecuritypolicies"},
+							Verbs:         []string{"use"},
+							ResourceNames: []string{policy.Name},
+						},
+					},
+				}
+
+				newRole.Annotations[podSecurityAnnotation] = "true"
+
+				role, err = m.roles.Create(newRole)
+
+				if err != nil {
+					return fmt.Errorf("error creating cluster role: %v", err)
+				}
+			}
+
+			// check if binding exists between service account and PSP
+			// if binding does not exist, create it
+			bindings, err = m.listBindings.List("", labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting bindings: %v", err)
+			}
+
+			bindingExists := false
+
+			for _, binding := range bindings {
+				if binding.Annotations[podSecurityAnnotation] == "true" {
+					bindingExists = true
+				}
+			}
+
+			if !bindingExists {
+				// create binding
+				newBinding := &rbac.ClusterRoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%v:%v-binding", obj.Name, policy.Name),
+						Namespace: namespaceName,
+						Annotations: map[string]string{
+							podSecurityAnnotation: "true",
+						},
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind: "ClusterRoleBinding",
+					},
+					RoleRef: rbac.RoleRef{
+						APIGroup: apiGroup,
+						Name:     policy.Name,
+						Kind:     "ClusterRole",
+					},
+					Subjects: []rbac.Subject{
+						{
+							Kind:      "ServiceAccount",
+							Name:      obj.Name,
+							Namespace: obj.Namespace,
+						},
+					},
+				}
+				_, err := m.bindings.Create(newBinding)
+
+				if err != nil {
+					logrus.Errorf("error creating binding '%v' error was '%v' ", newBinding, err)
+					return fmt.Errorf("error creating role binding: %v", err)
+				}
+			}
+
+			logrus.Infof("handle service account %v completed successfully", obj.Name)
+
+			return nil
+
+		})
+}
+
+type manager struct {
+	clusters     v3.ClusterLister
+	clustersInt  v3.ClusterInterface
+	templates    v3.PodSecurityPolicyTemplateLister
+	listPolicies v1beta1.PodSecurityPolicyLister
+	listBindings v12.ClusterRoleBindingLister
+	bindings     v12.ClusterRoleBindingInterface
+	policies     v1beta1.PodSecurityPolicyInterface
+	listRoles    v12.ClusterRoleLister
+	roles        v12.ClusterRoleInterface
+	namespaces   v13.NamespaceLister
+	projects     v3.ProjectLister
+}
```

Comment:
we are trying to standardize on a naming convention of
FooLister (for listers)
and Foos (for clients)
So, for example, your manager should look like
```
+	clusterRoleLister    v12.ClusterRoleLister
+	cluterRoles        v12.ClusterRoleInterface  // this is a client
+	namespaceLister   v13.NamespaceLister
+	projectLister    v3.ProjectLister
```